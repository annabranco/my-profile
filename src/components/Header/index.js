import React, { Component } from 'react';

export class Header extends Component {
  render() {
    const text = this.props.texts.Header;
    const language = this.props.language;

    return (
      <header className="header__outer">
        <div className="header__inner header__inner--languages">
          <img
            className={`header__flags ${
              language === 'en' ? 'header__flags--active' : ''
            }`}
            src="https://www.countryflags.io/us/flat/16.png"
            lang="en"
            onClick={this.props.changeLanguage}
            alt="English"
          />
          <img
            className={`header__flags ${
              language === 'es' ? 'header__flags--active' : ''
            }`}
            src="https://www.countryflags.io/es/flat/16.png"
            lang="es"
            onClick={this.props.changeLanguage}
            alt="Español"
          />
          <img
            className={`header__flags ${
              language === 'pt' ? 'header__flags--active' : ''
            }`}
            src="https://www.countryflags.io/br/flat/16.png"
            lang="pt"
            onClick={this.props.changeLanguage}
            alt="Português"
          />
          <img
            className={`header__flags hidden ${
              language === 'fr' ? 'header__flags--active' : ''
            }`}
            src="https://www.countryflags.io/fr/flat/16.png"
            lang="fr"
            onClick={this.props.changeLanguage}
            alt="Français"
          />
        </div>
        <div className="header__inner header__inner--title">
          <p className="header__text">{text[language].title}</p>
        </div>
        <div className="header__inner header__inner--version">
          <p className="header__text">v0.8.2</p>
          <p className="header__text-contact hidden">
            {text[language].contact}
          </p>
        </div>
      </header>
    );
  }
}
