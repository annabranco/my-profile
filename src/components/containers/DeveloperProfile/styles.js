import styled from 'styled-components';
import { rgba } from 'polished';
import Sea from '../../../images/bg/sea.gif';

export const SectionDeveloper = styled.section`
  position: relative;
  background-image: linear-gradient(
      ${rgba('#ffffff', 0.92)},
      ${rgba('#f0f1fc', 0.93)},
      ${rgba('#e6effb', 0.98)}
    ),
    url(${Sea});
  background-position: center;
  height: auto;
  overflow: hidden;
  padding-top: 50px;
`;

export const DeveloperInfoWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: auto;
`;
