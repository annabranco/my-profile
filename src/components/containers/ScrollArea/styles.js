import styled, { css } from 'styled-components';

export const ScrollAreaWrapper = styled.div`
  height: 100vh;
  overflow: scroll;

  ${props =>
    props.languageModalIsVisible &&
    css`
      overflow-y: hidden;
    `}
`;
