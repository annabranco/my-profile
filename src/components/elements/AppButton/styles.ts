import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueLight,
  colorBlack,
  fontTitle
} from '../../../styles/theme';

interface StyledButtonProps extends React.ComponentPropsWithoutRef<'button'> {
  buttonStyle: string;
}

export const StyledButton = styled.button<StyledButtonProps>`
  border: 1px solid ${colorBlack};
  border-radius: 15px 0;
  margin: 5px auto;
  background-image: linear-gradient(
    ${rgba(colorWhite, 0.2)},
    ${rgba(colorBlueLight, 0.6)}
  );
  width: 100px;
  padding: 5px 0;
  text-decoration: none;
  font-family: ${fontTitle};
  font-size: 1.2rem;
  cursor: pointer;
  box-shadow: 0 2px 4px 0 rgba($black, 0.4),
    inset 0 1px 5px 1px rgba($white, 0.7);

  &:hover {
    transform: translate(0, 1px);
    text-decoration: none;
    box-shadow: inset 0 2px 5px 1px rgba($black, 0.4);
  }

  ${({ buttonStyle }) => {
    return (
      buttonStyle &&
      css`
        color: ${buttonStyle};
      `
    );
  }}
`;
StyledButton.displayName = 'App Button';
