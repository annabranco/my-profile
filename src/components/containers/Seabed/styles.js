import styled, { css, keyframes } from 'styled-components';
import { rgba } from 'polished';
import {
  colorBlack,
  colorBlueDark,
  colorWhite,
  colorYellowBright,
  colorYellowPale,
  fontTitle,
  colorGrayDark,
  fontTitleAlt,
  colorRedBright
} from '../../../styles/theme';
import { NotDisplayed } from '../../../styles/global';

export const SeabedSection = styled.section`
  position: relative;
  background-image: linear-gradient(
      ${rgba('#bdd6f6', 0.96)},
      ${rgba('#abccf6', 0.95)},
      ${rgba('#ecf6bd', 0.9)}
    ),
    url('https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/bg/sea.gif');
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  height: 100vh;
  width: 100%;
`;
SeabedSection.displayName = '--- Seabed Section ---';

// ------- Floating animation

const FloatingAnimation = keyframes`
  from {
    transform : translate(-5px, 10px);
  }

  to {
    transform : translate(3px, -10px);
  }
`;

const FloatingSoft = css`
  animation-name: ${FloatingAnimation};
  animation-duration: 7s;
  animation-timing-function: ease;
  animation-iteration-count: infinite;
  animation-direction: alternate;
  animation-fill-mode: both;
`;

const Floating = css`
  z-index: 5;
  animation-name: float;
  animation-duration: 3s;
  animation-timing-function: ease;
  animation-iteration-count: infinite;
  animation-direction: alternate;
  animation-fill-mode: both;
`;

const FloatingDelay = css`
  animation-delay: 1s;
`;

// ---

export const MessageContainer = styled.div`
  border-radius: 80px 0;
  margin: 60px auto 0;
  background-color: rgba($white, 0.05);
  height: auto;
  width: 60vw;
  padding: 10px 50px;
  ${FloatingSoft};
`;
MessageContainer.displayName = '--- Message Container ---';

export const Message = styled.p`
  margin: 15px 0;
  font-family: $font__title;
  font-size: 1.2rem;
  line-height: 1.8;
  color: ${colorBlueDark};
  text-align: justify;
`;
Message.displayName = '--- Message ---';

export const MessageOnMobileDevices = styled(Message)`
  ${NotDisplayed};
`;
MessageOnMobileDevices.displayName = '--- Message ---';

const SeabedSubsectionContainer = styled.div`
  position: absolute;
  display: flex;
  width: 120px;
`;

export const SubsectionFormation = styled(SeabedSubsectionContainer)`
  bottom: 150px;
  left: 20vw;
  justify-content: flex-start;
  text-align: center;
  ${Floating};
  ${FloatingDelay};

  @media all and (min-width: 768px) {
    bottom: 33vh;
    left: auto;
    right: 30px;
    justify-content: flex-end;
    text-align: right;
  }
  ${({ hidden }) =>
    hidden &&
    css`
      display: hidden;
    `}
`;
SubsectionFormation.displayName = '--- Subsection Formation ---';

export const SubsectionOtherSkills = styled(SeabedSubsectionContainer)`
  top: 170px;
  bottom: initial;
  right: 33vw;
  justify-content: flex-end;
  text-align: center;

  @media all and (min-width: 768px) {
    top: initial;
    bottom: 33vh;
    left: 40px;
    right: auto;
    justify-content: flex-start;
    text-align: left;
  }
  ${({ hidden }) =>
    hidden &&
    css`
      display: hidden;
    `}
`;
SubsectionOtherSkills.displayName = '--- Subsection Other Skills ---';

export const Text = styled.p`
  opacity: 0.9;
  text-shadow: 0 0 3px ${colorBlueDark};
  font-family: sans-serif;
  font-size: 2.8rem;
  color: ${colorWhite};

  @media all and (min-width: 768px) {
    opacity: 0.7;
    text-shadow: 0 0 3px ${colorWhite};
    font-size: 3rem;
    color: ${rgba(colorWhite, 0.2)};
    ${({ active }) =>
      active &&
      css`
        @media all and (min-width: 768px) {
          text-shadow: 0 0 3px ${colorWhite};
          font-size: 3.2rem;
          color: ${rgba(colorYellowBright, 0.73)};
        }
      `}
  }
`;
Text.displayName = '--- Text ---';

export const HeroImage = styled.img`
  position: absolute;
  bottom: 40%;
  opacity: 0.6;
  width: 200px;
  transition: all ease 1s;
  ${NotDisplayed};

  @media all and (min-width: 768px) {
    display: block;
  }

  ${({ isSwimming, isGoingUp }) => {
    if (isSwimming) {
      return css`
        width: 250px;
        ${FloatingSoft}
      `;
    }
    if (isGoingUp) {
      return css`
        transform: rotate(290deg);
        width: 250px;
        transition: all 10s ease;
      `;
    }
    return css`
      ${FloatingSoft}
    `;
  }}
`;
HeroImage.displayName = '--- The Hero Image ---';

export const SeabedFloor = styled.div`
  position: absolute;
  bottom: 0;
  opacity: 0.05;
  background-image: url('https://raw.githubusercontent.com/annabranco/my-profile/master/src/images/bg/seabed.jpg');
  background-position: bottom;
  background-size: 100% 220%;
  height: 50vh;
  width: 100%;
`;
SeabedFloor.displayName = '--- Seabed Floor ---';

export const FloorText = styled.p`
  position: absolute;
  bottom: 20px;
  left: 17vw;
  transform: perspective(7rem) rotateX(16deg) skewX(20deg);
  width: 80%;
  text-shadow: 0 0 3px ${colorYellowBright};
  font-family: ${fontTitle};
  font-size: 9rem;
  line-height: 0.8;
  font-weight: bold;

  ${({ hidden }) =>
    hidden &&
    css`
      display: hidden;
    `}
`;
FloorText.displayName = '--- Floor Text ---';

export const GoBackText = styled.p`
  margin-top: 30vh;
  text-shadow: 0 0 2px ${colorBlack};
  font-family: ${fontTitle};
  font-size: 3rem;
  color: ${colorWhite};
  text-align: center;
`;
GoBackText.displayName = '--- Go Back Text ---';

export const ThinkingText = styled.p`
  position: absolute;
  bottom: 68%;
  opacity: 0.8;
  width: 180px;
  text-shadow: -1px 1px 6px ${colorBlack};
  font-style: italic;
  font-family: sans-serif;
  font-size: 1.2rem;
  color: ${colorYellowPale};
  text-align: center;

  ${({ side }) => {
    let leftValue = 'initial';
    let rightValue = '40px';

    if (side === 'on-left') {
      leftValue = '40px';
      rightValue = 'initial';
    }
    return css`
      left: ${leftValue};
      right: ${rightValue};
    `;
  }}
`;
ThinkingText.displayName = '--- Thinking Text ---';

export const TextFindSomething = styled.p`
  position: absolute;
  top: 20vh;
  left: 0;
  transform: translate(-50%, 0);
  width: 100%;
  text-shadow: 0 0 2px ${colorBlack};
  font-family: ${fontTitle};
  font-size: 2rem;
  color: ${colorWhite};
  text-align: center;

  @media all and (min-width: 768px) {
    top: 30vh;
    left: 50%;
    transform: translate(-50%, 0);
    font-size: 2.5rem;

    &:nth-of-type(2) {
      top: 37vh;
    }

    &:nth-of-type(3) {
      top: 44vh;
    }
  }

  &:nth-of-type(2) {
    top: 38vh;
  }
`;
TextFindSomething.displayName = '--- Text Find Something ---';

export const SeabedCloseButton = styled.button`
  z-index: 4;
  position: absolute;
  top: 10px;
  right: 5px;
  border: 1px solid $[colorRedDark];
  border-radius: 5px 0;
  background-image: linear-gradient(
    ${rgba(colorWhite, 0.2)},
    ${rgba(colorRedBright, 0.9)}
  );
  padding: 2px 5px;
  text-shadow: 0 0 1px ${rgba(colorGrayDark, 0.51)};
  font-family: ${fontTitleAlt};
  font-size: 0.7rem;
  color: ${colorBlack};
  cursor: pointer;
  box-shadow: 0 1px 2px 0 ${rgba(colorBlack, 0.4)},
    inset 0 1px 5px 1px ${rgba(colorWhite, 0.7)};

  @media all and (min-width: 321px) {
    border-radius: 8px 0;
    padding: 2px 10px;
    font-size: 1.2rem;
  }

  @media all and (min-width: 768px) {
    right: 10px;
    border-radius: 12px 0;
    padding: 3px 12px;
    font-size: 1.4rem;
  }

  &:hover {
    transform: translate(0, 1px);
    text-decoration: none;
    box-shadow: inset 0 2px 5px 1px ${rgba(colorBlack, 0.4)};
  }

  ${({ hidden }) =>
    hidden &&
    css`
      ${NotDisplayed}
    `}
`;
SeabedCloseButton.displayName = '--- Close Button ---';
