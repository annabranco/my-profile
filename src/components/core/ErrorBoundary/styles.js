import styled, { css } from 'styled-components';
import {
  colorWhite,
  colorBlueLight,
  colorBlack,
  colorGrayNormal,
  fontSubtitle,
  fontTitleAlt
} from '../../../styles/theme';
import { FullScreen, MainBackground } from '../../../styles/global';
import { SocialLinkButton } from '../../views/Social/styles';

export const ErrorSection = styled.div`
  ${MainBackground}
  ${FullScreen};
  margin: 0;
`;

export const ErrorTitle = styled.h2`
  margin-left: 0;
  width: auto;
  font-family: ${fontSubtitle};
  font-size: 2rem;
  text-align: left;
`;

export const ErrorNotifyWrapper = styled.div`
  position: relative;
  top: 50%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 80vw;
`;

export const ErrorSorryMessage = styled.p`
  margin: 0 0 20px;
  width: 80%;
  font-style: italic;
  font-family: ${fontSubtitle};
  font-size: 1.3rem;
  line-height: 1.5;
  color: ${colorBlueLight};
  text-align: left;
`;

export const ErrorDetailArea = styled.div`
  background: ${colorWhite};
  width: 60vw;
  padding: 10px 40px;
  box-sizing: border-box;
  border-radius: 8px 0;
  margin-top: 15px;
  box-shadow: 0 0 5px 2px ${colorWhite};

  &::before {
    content: 'ERROR:';
    font-size: 0.7rem;
    font-family: ${fontTitleAlt};
    color: ${colorGrayNormal};
  }
`;

export const ErrorDetailText = styled.p`
  font-size: 1rem;
`;

export const ErrorNotifyButton = styled(SocialLinkButton)`
  margin-top: 40px;
  position: fixed;
  bottom: 5%;
  opacity: 0;

  ${({ visible }) =>
    visible &&
    css`
      opacity: 1;
      transition: opacity 10s;
    `}
`;

export const ErrorNotifyButtonText = styled.p`
  position: relative;
  right: 0;
  margin: 4px 0 0 5px;
  font-family: ${fontTitleAlt};
  font-size: 1.3rem;
  color: ${colorBlack};
`;

export const ErrorNotifyButtonIcon = styled.i`
  display: inline;
  margin-left: 15px;
  vertical-align: middle;
  text-shadow: 0 0 1px ${colorBlack};
  font-size: 2rem;
`;
