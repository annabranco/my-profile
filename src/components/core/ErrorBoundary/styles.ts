import styled from 'styled-components';
import {
  colorWhite,
  colorBlueLight,
  colorBlack,
  colorGrayNormal,
  fontSubtitle,
  fontTitleAlt
} from '../../../styles/theme';
import { FullScreen, MainBackground } from '../../../styles/global';
import { LinkButton } from '../../elements/Social/styles';

export const DetailsText = styled.p`
  font-size: 1rem;
`;
DetailsText.displayName = 'DetailsText';

export const ErrorDetailsArea = styled.div`
  background: ${colorWhite};
  width: 100%;
  padding: 10px;
  border-radius: 8px 0;
  margin-top: 15px;
  box-shadow: 0 0 5px 2px ${colorWhite};

  &::before {
    content: 'ERROR:';
    font-size: 0.7rem;
    font-family: ${fontTitleAlt};
    color: ${colorGrayNormal};
  }

  @media all and (min-width: 768px) {
    width: 60vw;
    padding: 10px 40px;
  }
`;
ErrorDetailsArea.displayName = 'ErrorDetailsArea';

export const ErrorSection = styled.div`
  ${MainBackground}
  height: ${`${window.innerHeight}px`};
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;

  @media all and (min-width: 768px) {
    ${FullScreen};
    display: block;
  }
`;
ErrorSection.displayName = 'ErrorSection';

export const ErrorTitle = styled.h2`
  margin-left: 0;
  width: auto;
  font-family: ${fontSubtitle};
  font-size: 2rem;
  text-align: left;
`;
ErrorTitle.displayName = 'ErrorTitle';

export const NotificationArea = styled.div`
  position: relative;
  top: 5%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 80%;

  @media all and (min-width: 768px) {
    top: 50%;
    width: 80vw;
  }
`;
NotificationArea.displayName = 'NotificationArea';

interface NotifyButtonProps extends React.ComponentPropsWithoutRef<'a'> {
  visible?: boolean | null;
}

export const NotifyButton = styled(LinkButton)<NotifyButtonProps>`
  margin-top: 40px;
  position: fixed;
  bottom: 5%;
`;
NotifyButton.displayName = 'NotifyButton';

export const NotifyButtonIcon = styled.i`
  display: inline;
  margin-left: 15px;
  vertical-align: middle;
  text-shadow: 0 0 1px ${colorBlack};
  font-size: 2rem;
`;
NotifyButtonIcon.displayName = 'NotifyButtonIcon';

export const NotifyButtonText = styled.p`
  position: relative;
  right: 0;
  margin: 4px 0 0 5px;
  font-family: ${fontTitleAlt};
  font-size: 1.3rem;
  color: ${colorBlack};
`;
NotifyButtonText.displayName = 'NotifyButtonText';

export const SorryText = styled.p`
  margin: 0;
  width: 100%;
  font-style: italic;
  font-family: ${fontSubtitle};
  font-size: 1.1rem;
  line-height: 1.5;
  color: ${colorBlueLight};
  text-align: left;

  @media all and (min-width: 768px) {
    font-size: 1.3rem;
    margin: 0 0 20px;
    width: 80%;
  }
`;
SorryText.displayName = 'SorryText';
