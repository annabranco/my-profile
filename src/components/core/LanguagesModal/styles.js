import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueLight,
  colorBlueDark,
  colorBlack,
  colorGrayNormal,
  fontSubtitle,
  fontTitleAlt
} from '../../../styles/theme';
import { NotDisplayed } from '../../../styles/global';

const activeFlag = css`
  opacity: 1;
  cursor: default;
`;

export const Text = styled.p`
  margin: 5px 0;
  font-family: ${fontSubtitle};
  font-size: 0.8rem;
  text-align: center;

  @media all and (min-width: 560px) {
    font-size: 1rem;
  }
`;
Text.displayName = 'Text';

export const BackgroundOverlay = styled.div`
  z-index: 3;
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(
    to bottom,
    ${rgba(colorWhite, 0.6)},
    ${rgba(colorBlueDark, 0.4)},
    ${rgba(colorBlueDark, 0.6)},
    ${rgba(colorBlack, 0.9)}
  );
  height: 100vh;
  width: 100%;
`;
BackgroundOverlay.displayName = 'Background Overlay';

export const CheckBox = styled.input`
  margin-right: 20px;
`;
CheckBox.displayName = 'CheckBox';

export const CheckBoxArea = styled(Text)`
  margin: 30px 0 10px;
`;
CheckBoxArea.displayName = 'CheckBox Area';

export const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  border: 1px solid;
  border-radius: 40px 0;
  background-image: linear-gradient(
    to bottom left,
    ${colorWhite},
    #e6f4f4,
    ${colorBlueLight}
  );
  height: 280px;
  width: 340px;
  padding: 20px;
  box-shadow: 2px 2px 10px 2px ${rgba(colorGrayNormal, 0.7)};

  @media all and (min-width: 560px) {
    border-radius: 50px 0;
    height: 260px;
    width: 480px;
  }
`;
ModalWrapper.displayName = 'Modal Wrapper';

export const Flag = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  outline: none;
  margin: 10px;
  cursor: pointer;

  ${props =>
    !props.active &&
    css`
      ${NotDisplayed};
    `}
`;
Flag.displayName = 'Flag';

export const FlagsWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 70px;
  margin: 10px auto 0;

  @media all and (min-width: 560px) {
    margin: 20px auto 0;
  }
`;
FlagsWrapper.displayName = 'Flags Wrapper';

export const Image = styled.img`
  opacity: 0.5;
  margin-top: -10px;
  width: 100%;

  ${props =>
    props.lang === props.languageSelected &&
    css`
      ${activeFlag}
    `}
`;
Image.displayName = 'Image';

export const Label = styled.div`
  margin-bottom: 3px;
  font-family: ${fontTitleAlt};
  font-size: 0.8rem;
  text-align: center;
`;
Label.displayName = 'Label';
