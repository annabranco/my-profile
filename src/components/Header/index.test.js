import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Header from './index';
import appInfo from '../../../package.json';
import { setupMockProvider, getMockState } from '../../../testing/mocks';
import { CHANGE_LANGUAGE } from '../../constants';
import { Flag } from './styles';

const APP_VERSION = appInfo.version;

describe('< ErrorComponent >', () => {
  let MockProvider;
  let MOCK_STATE;
  let component;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    MOCK_STATE = getMockState({ language: 'en' });

    component = mount(
      <MockProvider>
        <Header hideForever />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(component).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should display the App title', () => {
    const { title } = MOCK_STATE.texts.header;
    expect(component.find('AppTitle').text()).toBe(title);
  });

  it('should display the App version', () => {
    expect(component.find('Version').text()).toBe(APP_VERSION);
  });

  it('should display a languages wrapper', () => {
    expect(component.find('LanguagesWrapper').exists()).toBe(true);
  });

  it('should display country flags for each active language from state', () => {
    const activeLanguages = [];

    MOCK_STATE.languages
      .filter(lang => lang.active)
      .forEach(lang => activeLanguages.push(lang.flagCode));

    const renderedFlags = component.find('Flag').children();

    renderedFlags.forEach((flagWrapper, index) => {
      const url = flagWrapper.children().first().props().src;
      const expectedUrl = `https://www.countryflags.io/${activeLanguages[index]}/flat/16.png`;

      expect(url).toBe(expectedUrl);
    });
    expect(renderedFlags).toHaveLength(activeLanguages.length);
  });

  it('should highlight the active language flag', () => {
    const flagEnglish = shallow(<Flag lang="en" languageSelected="en" />);
    const flagPortuguese = shallow(<Flag lang="pt" languageSelected="en" />);

    expect(toJson(flagEnglish)).toHaveStyleRule('opacity', '1');
    expect(toJson(flagPortuguese)).toHaveStyleRule(
      'opacity',
      expect.stringContaining('0.')
    );
  });

  it('should dispatch a CHANGE_LANGAUGE action when a flag is clicked', () => {
    const dispatchSpy = jest.spyOn(window, 'interceptedDispatch');
    const flag = component.find('Flag').children().last();
    const flagLanguage = flag.props().lang;

    flag.simulate('click');

    expect(dispatchSpy).toHaveBeenCalledTimes(1);
    expect(dispatchSpy).toHaveBeenCalledWith({
      payload: flagLanguage,
      type: CHANGE_LANGUAGE
    });
  });

  it('should change Title when state language changes', () => {
    const { title: englishTitle } = MOCK_STATE.texts.header;
    const { title: spanishhTitle } = getMockState({
      language: 'es'
    }).texts.header;

    expect(component.find('AppTitle').text()).toBe(englishTitle);

    const MockProviderWithStateChange = setupMockProvider({ language: 'es' })
      .MockProvider;
    const componentAfterLanguageChange = mount(
      <MockProviderWithStateChange>
        <Header hideForever />
      </MockProviderWithStateChange>
    );

    expect(componentAfterLanguageChange.find('AppTitle').text()).toBe(
      spanishhTitle
    );
  });
});
