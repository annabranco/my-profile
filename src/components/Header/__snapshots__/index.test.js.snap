// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ErrorComponent > matches the snapshot 1`] = `
<MockProvider>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Header
      hideForever={true}
    >
      <Header>
        <header
          className="gfNQNE"
        >
          <LanguagesWrapper>
            <div
              className="jHxTVq"
            >
              <Flag
                aria-label="English"
                key="en"
                lang="en"
                languageSelected="en"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  aria-label="English"
                  className="dooWXV"
                  lang="en"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <img
                    alt=""
                    src="https://www.countryflags.io/us/flat/16.png"
                  />
                </div>
              </Flag>
              <Flag
                aria-label="Español"
                key="es"
                lang="es"
                languageSelected="en"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  aria-label="Español"
                  className="jOqyQy"
                  lang="es"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <img
                    alt=""
                    src="https://www.countryflags.io/es/flat/16.png"
                  />
                </div>
              </Flag>
              <Flag
                aria-label="Português"
                key="pt"
                lang="pt"
                languageSelected="en"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  aria-label="Português"
                  className="jOqyQy"
                  lang="pt"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <img
                    alt=""
                    src="https://www.countryflags.io/br/flat/16.png"
                  />
                </div>
              </Flag>
            </div>
          </LanguagesWrapper>
          <AppTitle>
            <h2
              className="ja-dRuX"
            >
              Anna Branco's professional profile
            </h2>
          </AppTitle>
          <Version>
            <p
              className="jDAFlV"
            >
              0.14.1
            </p>
          </Version>
        </header>
      </Header>
    </Header>
  </Provider>
</MockProvider>
`;
